### PR TITLE
oidc extension added only if controlled-by-provisioner value is disabled

### PR DESCRIPTION
**Pull request by Disper**
**Description**
Adjusts feature parity of OIDC extension enablement between Provisioner and KIM. As for @tobiscr request:
> The Provisioner is enabling it only for new runtimes. In KIM, it looks like we enable it always - also for updates.
> Would KIM be able to detect if the RUntime was already created, or does it not know whether a Runtime is new or already exists?

Changes proposed in this pull request:

- extension applied only if `"operator.kyma-project.io/created-by-migrator": "true"` is not set
- ...
- ...

**Related issue(s)**
- https://github.com/kyma-project/infrastructure-manager/issues/381

### CODE REVIEW

**Comment by Disper** [comment]
After creating this PR I understand that this idea does not really make much sense. Existing clusters will anyway not have the OIDC extension enabled by KIM until we migrate the cluster:
1. We create Runtime CR for the cluster
2. Shoot will anyway then have `"kyma-project.io/controlled-by-provisioner": "false"` label

**Comment by Disper** [comment]
Implementation adjusted to rely on `operator.kyma-project.io/created-by-migrator` label instead.

File: internal/gardener/shoot/extender/oidc.go [hunk 1]
```diff
@@ -13,12 +13,25 @@ const (
...
+		setOIDCExtension(shoot)
+	}
 	setKubeAPIServerOIDCConfig(shoot, oidcConfig)
 
 	return nil
 }
 
+func CanEnableExtension(shoot *gardener.Shoot) bool {
+	canEnable := true
+	createdByMigrator := shoot.Labels["operator.kyma-project.io/created-by-migrator"]
```
Comment (akgalwas) [inline internal/gardener/shoot/extender/oidc.go:26]: `operator.kyma-project.io/created-by-migrator` is not available on the shoot.

**Review by akgalwas — CHANGES_REQUESTED** [review]
This will not work. The implementation looks for `operator.kyma-project.io/created-by-migrator` label on the shoot. The migrator app sets it on a **_runtime_**. There are two solutions:
- We can just check runtime for the label
- We can modify KIM to set this label on a shoot.

I think option number one is the best, as we don't need to mark shoot as migrated since Runtime CR contains this information already.

**Review by akgalwas — APPROVED** [review]

**Pull request merged**